src/helpers.js

```diff
@@ -218,29 +260,25 @@ const resolveUidForLink = (url, courseData, courseUidsLookup, pagePath) => {
   )
   const linkedCourse = courseUidsLookup[uid]
   if (linkedPage) {
-    // a course_page has been found for this UID
-    const linkPagePath = `${pathToChildRecursive(
-      path.join("courses", courseData["short_url"], "sections"),
-      linkedPage,
-      courseData
-    )}${getHugoPathSuffix(linkedPage, courseData)}`
-    return `${GETPAGESHORTCODESTART}${linkPagePath}${GETPAGESHORTCODEEND}`
+    // a page has been found for this UID
+    return pathLookup[uid] || null
   } else if (linkedFile) {
     // a course_file has been found for this UID
+    const parentUid = linkedFile["parent_uid"]
+
     if (linkedFile["file_type"] === "application/pdf") {
       // create a link to the generated PDF viewer page for this PDF file
-      const pdfPath = `${pagePath.replace("/_index.md", "/")}${
-        linkedFile["id"]
-      }`
-      return `${GETPAGESHORTCODESTART}${stripPdfSuffix(
-        pdfPath
-      )}${GETPAGESHORTCODEEND}`
+      const parent = pathLookup[parentUid]
+      if (parent) {
+        const pdfPath = path.join(parent, linkedFile["id"])
```

Comment:
I encountered an issue here with the course `18-06sc-linear-algebra-fall-2011`.  There are a couple of sections that have links to PDF's that were not functioning.  I dug into it and the HTML was being generated for them, so I was confused at first about what was wrong, but it seems the issue has to do with case sensitivity.  The URL's are being rendered into the markdown with uppercase letters, and the markdown files themselves have uppercase letters in the filenames.  However, it seems when Hugo generates the HTML it is lowercasing the filename (`MIT18_06SCF11_ex1.md` turns into a folder named `mit18_06scf11_ex1` with an `index.html` inside) and it only works if you access the URL lower case.

Reply:
This is an issue with the Hugo dev server but it would also most certainly be an issue with AWS hosted content as S3 / Cloudfront is most certainly case sensitive.

Reply:
Do you think this is serious enough to deal with in this PR or would you feel comfortable creating an issue for it? I did some investigation and it seems nontrivial to solve

Reply:
This could be solved in another PR, but in a way this PR actually causes the issue.  The links are generated properly using the `getpage` shortcode.  It's up to you though, I created an issue here: https://github.com/mitodl/ocw-to-hugo/issues/180

Reply:
This PR may cause that issue but unless it affects a large number of courses I think it would be simpler to deal with it in a separate PR

Reply:
Sounds good